### PR TITLE
REF: send/details to func component, move Address and Amount Inputs out of BlueComponents

### PR DESCRIPTION
**Pull request by limpbrains**
plan:
- e2e tests

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/46tuv3r5pqa7q3qeryc5udqc14

[download apk](http://filestorage.bluewallet.io:1488/limpbrains-new-send-1613470313.apk)

**Comment by Overtorment** [comment]
it would be best to make e2e first (for batch send as well), then do a refactor, and use e2e tests for acceptance.
its an important screen.

**Review by Overtorment — CHANGES_REQUESTED** [review]
hard to review such big changes. especially when a bunch of e2e fails.
maybe its better to start with separate PRs to increase coverage of that screen (don't forget batching)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/3am442n83y0ba3ugfdav4wbynm

[download apk](http://filestorage.bluewallet.io:1488/limpbrains-new-send-1613982805.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/6fzz3pcapjc6cy8nrmmv0mzb5c

[download apk](http://filestorage.bluewallet.io:1488/limpbrains-new-send-1613986610.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/uex2kt80gqawuk9qvw5pkauu2g

[download apk](http://filestorage.bluewallet.io:1488/limpbrains-new-send-1613988792.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/mvk3ttp1j72ay4ha5gtgrjm6n4

[download apk](http://filestorage.bluewallet.io:1488/limpbrains-new-send-1613997270.apk)

**Comment by limpbrains** [comment]
> hard to review such big changes. especially when a bunch of e2e fails.
> maybe its better to start with separate PRs to increase coverage of that screen (don't forget batching)

Will do in #1904

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/9mc3upjrd6vdtcrnr9c99n8h2c

[download apk](http://filestorage.bluewallet.io:1488/limpbrains-new-send-1614070439.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/4gfdxym1fuvgh2by3ax3r3fjjr

[download apk](http://filestorage.bluewallet.io:1488/limpbrains-new-send-1614072645.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/50h3gqk76ptad3vegrzyftzryw

[download apk](http://filestorage.bluewallet.io:1488/limpbrains-new-send-1614080387.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/mqk5kgepm0hmq7pmxy7j1ggnr4

[download apk](http://filestorage.bluewallet.io:1488/limpbrains-new-send-1614100909.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/3zc28a86tgz9xyk222k655t0jr

[download apk](http://filestorage.bluewallet.io:1488/limpbrains-new-send-1614247603.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/z7bg0x8tmjrxthe9rkgd2ybzkw

[download apk](http://filestorage.bluewallet.io:1488/limpbrains-new-send-1614251559.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/h1g28yn21txn14bq5nybzzrva4

[download apk](http://filestorage.bluewallet.io:1488/limpbrains-new-send-1614258636.apk)

File: screen/send/details.js [hunk 1]
```diff
@@ -1376,199 +1214,265 @@ export default class SendDetails extends Component {
...
+
+  return (
+    <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
+      <View style={[styles.root, stylesHook.root]}>
+        {/*
+          FIXME: <View style={styles.root} onLayout={this.onLayout} >
+          onLayout = e => {
+            this.setState({ width: e.nativeEvent.layout.width });
+          };
+        */}
```
Comment (limpbrains) [inline screen/send/details.js:1271]: @marcosrdz I've replaced this hack with width from `useWindowDimensions`. Can I keep it or should I use the original approach with onLayout={this.onLayout}?

Comment (marcosrdz) [reply screen/send/details.js:1271]: onLayout is needed for desktop and tablets. useWindowDimensions does not work well with those devices. It returns inaccurate dimensions

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/b5r7na26g3c65ekw2yrdu0r438

[download apk](http://filestorage.bluewallet.io:1488/limpbrains-new-send-1614267957.apk)

**Comment by limpbrains** [comment]
Ready for review and merge

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/z2vh6v38b14v6r3cgfn59mc29m

[download apk](http://filestorage.bluewallet.io:1488/limpbrains-new-send-1614270470.apk)

**Comment by marcosrdz** [comment]
@limpbrains  conflicts

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/3fuzcadhpdjdxgzqhdecek916c

[download apk](http://filestorage.bluewallet.io:1488/limpbrains-new-send-1614370515.apk)

File: screen/send/details.js [hunk 2]
```diff
@@ -55,594 +52,403 @@ const scanqr = require('../../helpers/scan-qr');
...
+      Keyboard.removeListener('keyboardDidShow', _keyboardDidShow);
+      Keyboard.removeListener('keyboardDidHide', _keyboardDidHide);
+    };
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
+  useEffect(() => {
+    // check if we have a suitable wallet
+    const suitable = wallets.filter(wallet => wallet.chain === Chain.ONCHAIN && wallet.allowSend());
+    if (suitable.length === 0) {
+      Alert.alert(loc.send.details_wallet_before_tx);
```
Comment (marcosrdz) [inline screen/send/details.js:125]: Alerts should have a title instead of using the message as the title. Especially for macOS.

**Review by marcosrdz — CHANGES_REQUESTED** [review]

**Review by marcosrdz — CHANGES_REQUESTED** [review]
send/details.js screen is not listening for deeplinks. (such as deepLinkPSBT). 
Perhaps addresses should be a useRoute.params instead of useState? That way the screen updates if a bitcoin address is deeplinked in while this modal is open

File: screen/send/details.js (hunk 2 above)
Comment (limpbrains) [reply screen/send/details.js]: fixed

File: screen/send/details.js (hunk 1 above)
Comment (limpbrains) [reply screen/send/details.js]: fixed

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/kv2015cap1d5nm3m72vmxnx29c

[download apk](http://filestorage.bluewallet.io:1488/limpbrains-new-send-1614498589.apk)

**Comment by limpbrains** [comment]
> send/details.js screen is not listening for deeplinks. (such as deepLinkPSBT).
> Perhaps addresses should be a useRoute.params instead of useState? That way the screen updates if a bitcoin address is deeplinked in while this modal is open

I can see `deepLinkPSBT` is used in `psbtWithHardwareWallet.js`. So when you open psbt file BW launches `PsbtWithHardwareWallet`. How does this relates with sendDetails? 

If this is some kind of new feature, let's add it in separate PR, because the longer we will develop this one, the harder it will be to sync all new changes.

**Review by marcosrdz — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -48,7 +48,7 @@
     "e2e:debug-test": "detox test -c android.emu.debug",
     "e2e:debug": "(test -f android/app/build/outputs/apk/debug/app-debug.apk && test -f android/app/build/outputs/apk/androidTest/debug/app-debug-androidTest.apk) || npm run e2e:debug-build; npm run e2e:debug-test",
     "e2e:release-build": "npx detox build -c android.emu.release",
-    "e2e:release-test": "detox test -c android.emu.release --record-videos all --take-screenshots all --headless",
+    "e2e:release-test": "detox test -c android.emu.release --record-videos all --take-screenshots all --headless --loglevel trace",
```
Comment (Overtorment) [inline package.json:51]: is it gona stay with us forever?

**Review by Overtorment — APPROVED** [review]
gj. its nice to have refactoring thats also covered by tests

**Comment by GladosBlueWallet** [comment]
Unbelievable. You, [subject name here], must be the pride of [subject hometown here]!

**Comment by Overtorment** [comment]
merged. lets continue testing on real builds

**Pull request merged**